pkg/subroutines/creator_test.go

```diff
@@ -0,0 +1,354 @@
+				openFGAServiceClientMock.EXPECT().Write(ctx, in).
+					Return(out, nil).
+					Once()
+			},
+		},
+		{
+			name: "should_not_delete_owner_if_creator_is_nil",
+			account: v1alpha1.Account{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-account",
+					Namespace: "openmfp-system",
+				},
+				Status: v1alpha1.AccountStatus{
+					Namespace: &namespace,
+				},
+			},
+			in: &openfgav1.WriteRequest{
+				StoreId: "1",
+				Deletes: &openfgav1.WriteRequestDeletes{
+					TupleKeys: []*openfgav1.TupleKeyWithoutCondition{
+						{
+							Object:   "account:test-account",
+							Relation: "owner",
+							User:     "user:test-creator",
+						},
+					},
+				},
+			},
+			setupMocks: func(ctx context.Context, in *openfgav1.WriteRequest, out *openfgav1.WriteResponse, openFGAServiceClientMock *mocks.OpenFGAServiceClient, k8ServiceMock *mocks.K8Service) {
+			},
+		},
+		{
+			name:        "handle_write_error",
+			expectError: true,
+			account: v1alpha1.Account{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-account",
+					Namespace: "openmfp-system",
+				},
+				Spec: v1alpha1.AccountSpec{
+					Creator: &creator,
+				},
+				Status: v1alpha1.AccountStatus{
+					Namespace: &namespace,
+				},
+			},
+			in: &openfgav1.WriteRequest{
+				StoreId: "5",
+				Deletes: &openfgav1.WriteRequestDeletes{
+					TupleKeys: []*openfgav1.TupleKeyWithoutCondition{
+						{
+							Object:   "account:test-account",
+							Relation: "owner",
+							User:     "user:test-creator",
+						},
+					},
+				},
+			},
+			setupMocks: func(ctx context.Context, in *openfgav1.WriteRequest, out *openfgav1.WriteResponse, openFGAServiceClientMock *mocks.OpenFGAServiceClient, k8ServiceMock *mocks.K8Service) {
+				k8ServiceMock.EXPECT().GetFirstLevelAccountForNamespace(ctx, "openmfp-system").Return(&v1alpha1.Account{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "tenant5",
+						Namespace: "openmfp-system",
+					},
+					Spec: v1alpha1.AccountSpec{
+						Creator: &creator,
+					},
+					Status: v1alpha1.AccountStatus{
+						Namespace: &namespace,
+					},
+				}, nil)
+				openFGAServiceClientMock.EXPECT().ListStores(ctx, mock.Anything).Return(&openfgav1.ListStoresResponse{
+					Stores: []*openfgav1.Store{{Id: "5", Name: "tenant-tenant5"}},
+				}, nil).Once()
+				openFGAServiceClientMock.EXPECT().Write(ctx, in).
+					Return(nil, assert.AnError).
+					Once()
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+
+			openFGAServiceClientMock := &mocks.OpenFGAServiceClient{}
+			k8ServiceMock := mocks.NewK8Service(t)
+			if test.setupMocks != nil {
+				out := test.out
+				if out == nil {
+					out = &openfgav1.WriteResponse{}
+				}
+				test.setupMocks(test.ctx, test.in, out, openFGAServiceClientMock, k8ServiceMock)
+			}
+
+			routine := subroutines.NewCreatorSubroutine(openFGAServiceClientMock, k8ServiceMock, "openmfp-system")
+			_, err := routine.Finalize(test.ctx, &test.account)
```

Comment:
🟢 check for the operator result, and also if sentry and retry is set. 
We faced severe issues already, if retry is false